### ISSUE TITLE
Left button for PIN entry not functioning on Windows USB for TTGO Jade wallet DIY

### ISSUE THREAD

**Issue by bayfitt** [Blockstream/Jade#62]
Description:
I would like to report a bug for the TTGO Jade wallet DIY. I have tested the wallet on USB on Windows, Linux, and Android. When I went to enter my PIN, the left button on the TTGO did not work on Windows, while the right button was functional. I have a video recording of the issue, which I can share with the developers. I consistently experienced this issue even after restarting both the wallet and the computer on Windows. On the other hand, on Linux and Android on USB, everything worked fine. Additionally, Bluetooth on Android worked as expected, and both buttons functioned correctly when entering my PIN. 

Details:
- Issue: The left button on the TTGO did not work when entering my PIN on Windows
- Reproduction Steps: 
    1. Power on the TTGO Jade wallet DIY
    2. Connect it to a Windows computer via USB
    3. Try to enter your PIN using the left button
- Expected Result: The left button should work correctly, allowing me to enter my PIN.
- Actual Result: The left button does not work on Windows, preventing me from entering my PIN.
- Video Evidence: I have a video recording of the issue, which I can share with the developers.
- Firmware version: 0.1.47-19-ga509904
- Secure Boot enabled: Yes

Additional Notes:
- The issue was not present on Linux and Android on USB.
- Bluetooth on Android worked as expected, and both buttons functioned correctly when entering my PIN. 
- I would appreciate it if the developers could investigate this issue and provide a fix as soon as possible.

**Comment by bayfitt** [comment]
Uploading 1_5064448508560409232.mp4 …

Uploading 1_5064448508560409231.mp4 …

**Comment by bayfitt** [comment]
Uploading 1_5064448508560409232.mp4 …

Uploading 1_5064448508560409231.mp4 …

**Comment by bayfitt** [comment]
https://user-images.githubusercontent.com/26312519/235023414-4ffc3ad7-0570-41b2-a3c6-fb125aa2d82f.mp4

**Comment by bayfitt** [comment]
https://user-images.githubusercontent.com/26312519/235023914-9346cd3d-23ee-4c4e-b96f-409a112bdc43.mp4

https://user-images.githubusercontent.com/26312519/235023943-e093a299-16df-4911-be84-f6a65dcbde7a.mp4

**Comment by jpph** [comment]
As a temporary workaround, you can use electrum for windows to unlock the ttgo display, then close electrum, then launch Blockstream Green for windows.

**Comment by bayfitt** [comment]
My concern was not about it not working in windows. I can use Linux or Android. My concern is that it works differently, which means there is something going inside the computer from that wallet. Why should it work differently in different OSes, if it is supposed to be isolated? To me, this is a security concern.

**Comment by jpph** [comment]
As per cryptoguide told me there is some simplified input mode when connected to other software than  green wallet. Seems that android and Linux version are using the simplified mode

**Comment by JamieDriver** [comment]
This sounds v odd to me - you are not re-flashing the device between using on windows and linux, right ?
- You plug it into linux, and can use both buttons and unlock the device with PIN
- You plug the same device/same fw etc. into windows, and the left button does not work ?

This is certainly not expected.  
If I had to hazard a guess I'd say that maybe there is some pin/interrupt conflict that windows is triggering which is somehow disrupting detecting the button press.  ('pin' as in physical pin on the board, not as in 'personal id number'!)

**Comment by JamieDriver** [comment]
So you plug a lilygo t-display into Green Desktop on Windows, and the left button does not work ?
(iirc the right and left buttons should scroll the numbers up/down, and you press both together to move to 'next digit' ?)

We will certainly investigate - as you say the unit buttons should not be affected by whatever happens to be on the other end of the USB cable!

**Comment by jpph** [comment]
I have same behavior that on the video. At first digit left button act as ''both button'', then at second digit it stop working. And both button stop working too. Very strange behavior

**Comment by JamieDriver** [comment]
Looks like something specific to Green Desktop (on windows) rather than an issue with the Jade ttgo fw per se.  (eg. setting some usb/serial flags badly, voltage on wrong pins, etc.).
We'll investigate further - thanks for bringing it to our attention!

**Comment by sandman21vs** [comment]
I'm having exactly the same problem, the tip to connect the electrum was good, I hadn't thought of that, thanks.
I hope they fix this bug soon and it's not a security breach

**Comment by JamieDriver** [comment]
Are you running the fw with BLE/radio enabled ?

**Comment by sandman21vs** [comment]
I'm running directly connected to the cable

**Comment by JamieDriver** [comment]
yy, but I was wondering if Bluetooth was running/enabled, even if youre not connected by Bluetooth - just in case it was relevant ...  (if 'Yes' - does it make any difference if you disable it ?)  
This is all info I can pass to the desktop guys.

**Comment by jpph** [comment]
For myself, I have the issue with cable connected and Bluetooth disabled.

**Comment by sandman21vs** [comment]
Both with Bluetooth on and off, jade diy does not work in the blockstream app (but only in the Green app, in other apps it works perfectly)

**Comment by JamieDriver** [comment]
Cheers.  I'll pass the details along to the windows/desktop app guys.  
We'll also check the build/button config to see if we've got usb/serial signals and button signals getting tangled.

Are you all on lilygo-tdisplay units ?

**Comment by sandman21vs** [comment]
yes we are using liligo tdisplay, in a few days my camera arrives and then I'm going to make a diy jade with it too

**Comment by adri76** [comment]
I can confirm that I have the same issue. It is only a problem when using Green wallet via USB. Bluetooth works fine.

**Comment by Sandmann21s** [comment]
> Posso confirmar que tenho o mesmo problema. É apenas um problema ao usar a carteira verde via USB. Bluetooth funciona bem.


I can't use Jade via BT on Windows, but on Linux the blockstream green app works well via cable 

Another problem is that the battery indicator does not work showing the device charge

**Comment by JamieDriver** [comment]
> Another problem is that the battery indicator does not work showing the device charge

Yes, that hw board has no chip telling us that sort of information.

**Comment by Sandmann21s** [comment]
![image](https://github.com/user-attachments/assets/7de72ce4-8bfb-4071-ae23-3fb9a134d568)
This issue has been resolved, and now the button works correctly on the Jade DIY. Thank you for the fix!

**Comment by Sandmann21s** [comment]
Through trial and error I discovered that if you format your DIY Jade, open the Green app in Windows, connect the Jade, and do the setup it starts to work, but there is no way to connect a Jade that you did not setup in Windows, (the bug does not persist if you connect the Jade after the Green app opens, but if the Jade is not configured Windows does not try to connect to it, when it tries to connect it gives an error lol)

Just to comment, I tested it with an M5stack C2Plus and it works normally, apparently it is only the TTGO T Display that has this problem, the sad part is that this is the most used DIY device to make DIY Jade

**Comment by adri76** [comment]
I tried again, but could not get it to work with green wallet.
It works with other wallet apps, but not green wallet. (no way to enter the pin)
The point of a hardware wallet is to securely store your keys, and I would not trust a buggy setup to store anything of value, so it is nice to use as a proof of concept, but it defeats the object of a diy wallet.

**Comment by Sandmann21s** [comment]
What's interesting is that this issue only occurs in the Windows version. I believe it would be helpful to open an issue on Green's GitHub to get it resolved because it works fine on Android, macOS, and Linux—only the Windows version is affected.

**Comment by sandman21vs** [comment]
The previously reported bug has changed at this time, I don't know exactly what is happening but Jade now clicks both blocks alone once when the green wallet is opened and after that it keeps clicking the 

https://github.com/user-attachments/assets/b71c0922-a600-4f1c-ba2b-7a5270377227

top button infinitely and does what appears in the video above

**Comment by jpph** [comment]
well if problem can be solved on the hardware and root cause has been found, would be nice to implement a fix inside green for windows. Remember that there is no issue with electrum or sparrow on windows

**Comment by sandman21vs** [comment]
Watch the video at 31:30 seconds; I have an explanation here on how to solve this hardware problem. 

https://youtu.be/nCkqMz7YTQc
